src/server/db_slice.cc

```diff
@@ -1338,4 +1338,27 @@ void DbSlice::ResetUpdateEvents() {
   events_.update = 0;
 }
 
+void DbSlice::TrackKeys(const facade::Connection::WeakRef& conn,
+                        const std::vector<std::string_view>& keys) {
+  if (conn.IsExpired()) {
+    DVLOG(2) << "Connection expired, exiting TrackKey function.";
+    return;
+  }
+
+  DVLOG(2) << "Start tracking keys for client ID: " << conn.GetClientId()
+           << " with thread ID: " << conn.Thread();
+  for (auto key : keys) {
+    if (client_tracking_map_.find(key) == client_tracking_map_.end()) {
+      DVLOG(2) << "The key " << key << " has not been tracked by any client.";
+      DVLOG(2) << "Creating a new entry in the tracking table for this key.";
+      absl::flat_hash_set<facade::Connection::WeakRef, Hash> tracking_client_set{conn};
+      client_tracking_map_[key] = tracking_client_set;
+    } else {
+      DVLOG(2) << "The key " << key << " exists in the client tracking table.";
+      DVLOG(2) << "Inserting client ID " << conn.GetClientId() << " into its tracking client set: ";
+      client_tracking_map_[key].insert(conn);
```

Comment:
I know you have different output for now, but at the end you don't need this if  

map[key].insert() works even if key is not in map